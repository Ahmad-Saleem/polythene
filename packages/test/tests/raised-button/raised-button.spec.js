/* global describe, it, expect */
import m from "mithril";
import { tidy, defaultHtmlTidyOptions } from "mithril-jest";
import { tests } from "./tests.js";
import { runSnapshots } from "../../scripts/snapshots";
import raisedButton from "polythene-raised-button";

runSnapshots(tests);

describe("Raised button component", () => {
  it("no options", () => {
    const cmp = m(raisedButton);
    const html = tidy(cmp, {
      ...defaultHtmlTidyOptions,
      wrap: false,
      indent: false
    });
    expect(html).toContain("<a tabindex=\"0\" class=\"pe-button pe-text-button pe-raised-button        \"></a>");
  });
  it("child", () => {
    const cmp = m(raisedButton, {}, m("span", "Child"));
    const html = tidy(cmp);
    expect(html).toContain("<span>Child</span>");
  });
  it("option content", () => {
    const cmp = m(raisedButton, {content: m("span", "Content")});
    const html = tidy(cmp);
    expect(html).toContain("<span>Content</span>");
  });
  it("option id", () => {
    const cmp = m(raisedButton, {id: "id-x"});
    const html = tidy(cmp);
    expect(html).toContain("id=\"id-x\"");
  });
  it("option class", () => {
    const cmp = m(raisedButton, {class: "class-x"});
    const html = tidy(cmp);
    expect(html).toContain("class-x\">");
  });
  it("option element", () => {
    const cmp = m(raisedButton, {element: "button"});
    const html = tidy(cmp);
    expect(html).toContain("<button ");
    expect(html).toContain("</button>");
  });
  it("option tabindex", () => {
    const cmp = m(raisedButton, {tabindex: 3});
    const html = tidy(cmp);
    expect(html).toContain("tabindex=\"3\"");
  });
  it("option before", () => {
    const cmp = m(raisedButton, {before: m("span", "Before")});
    const html = tidy(cmp);
    expect(html).toContain("<span>Before</span>");
  });
  it("option after", () => {
    const cmp = m(raisedButton, {after: m("span", "After")});
    const html = tidy(cmp);
    expect(html).toContain("<span>After</span>");
  });
});
