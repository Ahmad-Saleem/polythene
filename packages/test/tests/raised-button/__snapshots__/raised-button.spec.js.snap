exports[`test Option: animateOnTap (false) 1`] = `
"<a tabindex=\"0\" class=\"pe-button pe-text-button pe-raised-button\">
<div class=\"pe-button__content\">
  <div class=\"pe-shadow pe-shadow--animated\">
    <div class=\"pe-shadow__bottom pe-shadow--z-1\"></div>
    <div class=\"pe-shadow__top pe-shadow--z-1\"></div>
  </div>
  <div class=\"pe-ripple pe-ripple--constrained\">
    <div class=\"pe-ripple__mask\">
      <div class=\"pe-ripple__waves\"></div>
    </div>
  </div>
  <div class=\"pe-button__wash\"></div>
  <div class=\"pe-button__focus\"></div>
  <div class=\"pe-button__label\">
    Don\'t animate shadow
  </div>
</div></a>
"
`;

exports[`test Option: disabled (false) 1`] = `
"<a tabindex=\"0\" class=\"pe-button pe-text-button pe-raised-button\">
<div class=\"pe-button__content\">
  <div class=\"pe-shadow pe-shadow--animated\">
    <div class=\"pe-shadow__bottom pe-shadow--z-1\"></div>
    <div class=\"pe-shadow__top pe-shadow--z-1\"></div>
  </div>
  <div class=\"pe-ripple pe-ripple--constrained\">
    <div class=\"pe-ripple__mask\">
      <div class=\"pe-ripple__waves\"></div>
    </div>
  </div>
  <div class=\"pe-button__wash\"></div>
  <div class=\"pe-button__focus\"></div>
  <div class=\"pe-button__label\">
    Not disabled
  </div>
</div></a>
"
`;

exports[`test Option: disabled (true) 1`] = `
"<a tabindex=\"-1\" disabled class=
\"pe-button pe-text-button pe-raised-button pe-button--disabled\">
<div class=\"pe-button__content\">
  <div class=\"pe-button__label\">
    Disabled
  </div>
</div></a>
"
`;

exports[`test Option: disabled -- dark theme 1`] = `
"<a tabindex=\"-1\" disabled class=
\"pe-button pe-text-button pe-raised-button pe-button--disabled\">
<div class=\"pe-button__content\">
  <div class=\"pe-button__label\">
    Label
  </div>
</div></a>
"
`;

exports[`test Option: events (onclick) 1`] = `
"<div>
  onclick called: 0
</div><a tabindex=\"0\" class=
\"pe-button pe-text-button pe-raised-button\">
<div class=\"pe-button__content\">
  <div class=\"pe-shadow pe-shadow--animated\">
    <div class=\"pe-shadow__bottom pe-shadow--z-1\"></div>
    <div class=\"pe-shadow__top pe-shadow--z-1\"></div>
  </div>
  <div class=\"pe-ripple pe-ripple--constrained\">
    <div class=\"pe-ripple__mask\">
      <div class=\"pe-ripple__waves\"></div>
    </div>
  </div>
  <div class=\"pe-button__wash\"></div>
  <div class=\"pe-button__focus\"></div>
  <div class=\"pe-button__label\">
    Button
  </div>
</div></a>
"
`;

exports[`test Option: inactive (false) 1`] = `
"<a tabindex=\"0\" class=\"pe-button pe-text-button pe-raised-button\">
<div class=\"pe-button__content\">
  <div class=\"pe-shadow pe-shadow--animated\">
    <div class=\"pe-shadow__bottom pe-shadow--z-1\"></div>
    <div class=\"pe-shadow__top pe-shadow--z-1\"></div>
  </div>
  <div class=\"pe-ripple pe-ripple--constrained\">
    <div class=\"pe-ripple__mask\">
      <div class=\"pe-ripple__waves\"></div>
    </div>
  </div>
  <div class=\"pe-button__wash\"></div>
  <div class=\"pe-button__focus\"></div>
  <div class=\"pe-button__label\">
    Not inactive
  </div>
</div></a>
"
`;

exports[`test Option: inactive (true) 1`] = `
"<a tabindex=\"-1\" class=
\"pe-button pe-text-button pe-raised-button pe-button--inactive\">
<div class=\"pe-button__content\">
  <div class=\"pe-shadow pe-shadow--animated\">
    <div class=\"pe-shadow__bottom pe-shadow--z-1\"></div>
    <div class=\"pe-shadow__top pe-shadow--z-1\"></div>
  </div>
  <div class=\"pe-ripple pe-ripple--constrained\">
    <div class=\"pe-ripple__mask\">
      <div class=\"pe-ripple__waves\"></div>
    </div>
  </div>
  <div class=\"pe-button__wash\"></div>
  <div class=\"pe-button__focus\"></div>
  <div class=\"pe-button__label\">
    Inactive
  </div>
</div></a>
"
`;

exports[`test Option: ink (false) 1`] = `
"<a tabindex=\"0\" class=\"pe-button pe-text-button pe-raised-button\">
<div class=\"pe-button__content\">
  <div class=\"pe-shadow pe-shadow--animated\">
    <div class=\"pe-shadow__bottom pe-shadow--z-1\"></div>
    <div class=\"pe-shadow__top pe-shadow--z-1\"></div>
  </div>
  <div class=\"pe-button__wash\"></div>
  <div class=\"pe-button__focus\"></div>
  <div class=\"pe-button__label\">
    No ink
  </div>
</div></a>
"
`;

exports[`test Option: label -- dark theme (should be app's primary color) 1`] = `
"<a tabindex=\"0\" class=\"pe-button pe-text-button pe-raised-button\">
<div class=\"pe-button__content\">
  <div class=\"pe-shadow pe-shadow--animated\">
    <div class=\"pe-shadow__bottom pe-shadow--z-1\"></div>
    <div class=\"pe-shadow__top pe-shadow--z-1\"></div>
  </div>
  <div class=\"pe-ripple pe-ripple--constrained\">
    <div class=\"pe-ripple__mask\">
      <div class=\"pe-ripple__waves\"></div>
    </div>
  </div>
  <div class=\"pe-button__wash\"></div>
  <div class=\"pe-button__focus\"></div>
  <div class=\"pe-button__label\">
    Label
  </div>
</div></a>
"
`;

exports[`test Option: label 1`] = `
"<a tabindex=\"0\" class=\"pe-button pe-text-button pe-raised-button\">
<div class=\"pe-button__content\">
  <div class=\"pe-shadow pe-shadow--animated\">
    <div class=\"pe-shadow__bottom pe-shadow--z-1\"></div>
    <div class=\"pe-shadow__top pe-shadow--z-1\"></div>
  </div>
  <div class=\"pe-ripple pe-ripple--constrained\">
    <div class=\"pe-ripple__mask\">
      <div class=\"pe-ripple__waves\"></div>
    </div>
  </div>
  <div class=\"pe-button__wash\"></div>
  <div class=\"pe-button__focus\"></div>
  <div class=\"pe-button__label\">
    Label
  </div>
</div></a>
"
`;

exports[`test Option: raised (with option z: 2) 1`] = `
"<a tabindex=\"0\" class=\"pe-button pe-text-button pe-raised-button\">
<div class=\"pe-button__content\">
  <div class=\"pe-shadow pe-shadow--animated\">
    <div class=\"pe-shadow__bottom pe-shadow--z-2\"></div>
    <div class=\"pe-shadow__top pe-shadow--z-2\"></div>
  </div>
  <div class=\"pe-ripple pe-ripple--constrained\">
    <div class=\"pe-ripple__mask\">
      <div class=\"pe-ripple__waves\"></div>
    </div>
  </div>
  <div class=\"pe-button__wash\"></div>
  <div class=\"pe-button__focus\"></div>
  <div class=\"pe-button__label\">
    Raised to 2
  </div>
</div></a>
"
`;

exports[`test Option: raised (with option z: 5) 1`] = `
"<a tabindex=\"0\" class=\"pe-button pe-text-button pe-raised-button\">
<div class=\"pe-button__content\">
  <div class=\"pe-shadow pe-shadow--animated\">
    <div class=\"pe-shadow__bottom pe-shadow--z-5\"></div>
    <div class=\"pe-shadow__top pe-shadow--z-5\"></div>
  </div>
  <div class=\"pe-ripple pe-ripple--constrained\">
    <div class=\"pe-ripple__mask\">
      <div class=\"pe-ripple__waves\"></div>
    </div>
  </div>
  <div class=\"pe-button__wash\"></div>
  <div class=\"pe-button__focus\"></div>
  <div class=\"pe-button__label\">
    Raised to 5
  </div>
</div></a>
"
`;

exports[`test Option: selected 1`] = `
"<a tabindex=\"0\" class=
\"pe-button pe-text-button pe-raised-button pe-button--selected\">
<div class=\"pe-button__content\">
  <div class=\"pe-shadow pe-shadow--animated\">
    <div class=\"pe-shadow__bottom pe-shadow--z-1\"></div>
    <div class=\"pe-shadow__top pe-shadow--z-1\"></div>
  </div>
  <div class=\"pe-ripple pe-ripple--constrained\">
    <div class=\"pe-ripple__mask\">
      <div class=\"pe-ripple__waves\"></div>
    </div>
  </div>
  <div class=\"pe-button__wash\"></div>
  <div class=\"pe-button__focus\"></div>
  <div class=\"pe-button__label\">
    Selected
  </div>
</div></a>
"
`;

exports[`test Option: style (colors) 1`] = `
"<a tabindex=\"0\" class=\"pe-button pe-text-button pe-raised-button\">
<div style=
\"background-color: rgb(239, 108, 0); color: rgb(255, 255, 255);\"
class=\"pe-button__content\">
  <div class=\"pe-shadow pe-shadow--animated\">
    <div class=\"pe-shadow__bottom pe-shadow--z-1\"></div>
    <div class=\"pe-shadow__top pe-shadow--z-1\"></div>
  </div>
  <div class=\"pe-ripple pe-ripple--constrained\">
    <div class=\"pe-ripple__mask\">
      <div class=\"pe-ripple__waves\"></div>
    </div>
  </div>
  <div class=\"pe-button__wash\"></div>
  <div class=\"pe-button__focus\"></div>
  <div class=\"pe-button__label\">
    Styled
  </div>
</div></a>
"
`;

exports[`test Option: url (with oncreate) 1`] = `
"<a tabindex=\"0\" href=\"#!/shadow\" class=
\"pe-button pe-text-button pe-raised-button\">
<div class=\"pe-button__content\">
  <div class=\"pe-shadow pe-shadow--animated\">
    <div class=\"pe-shadow__bottom pe-shadow--z-1\"></div>
    <div class=\"pe-shadow__top pe-shadow--z-1\"></div>
  </div>
  <div class=\"pe-ripple pe-ripple--constrained\">
    <div class=\"pe-ripple__mask\">
      <div class=\"pe-ripple__waves\"></div>
    </div>
  </div>
  <div class=\"pe-button__wash\"></div>
  <div class=\"pe-button__focus\"></div>
  <div class=\"pe-button__label\">
    Go to /#/shadow
  </div>
</div></a>
"
`;

exports[`test Option: wash (false) 1`] = `
"<a tabindex=\"0\" class=\"pe-button pe-text-button pe-raised-button\">
<div class=\"pe-button__content\">
  <div class=\"pe-shadow pe-shadow--animated\">
    <div class=\"pe-shadow__bottom pe-shadow--z-1\"></div>
    <div class=\"pe-shadow__top pe-shadow--z-1\"></div>
  </div>
  <div class=\"pe-ripple pe-ripple--constrained\">
    <div class=\"pe-ripple__mask\">
      <div class=\"pe-ripple__waves\"></div>
    </div>
  </div>
  <div class=\"pe-button__focus\"></div>
  <div class=\"pe-button__label\">
    No wash
  </div>
</div></a>
"
`;

exports[`test Themed button (should be red) 1`] = `
"<a tabindex=\"0\" class=
\"pe-button pe-text-button pe-raised-button tests-raised-button-themed-button\">
<div class=\"pe-button__content\">
  <div class=\"pe-shadow pe-shadow--animated\">
    <div class=\"pe-shadow__bottom pe-shadow--z-1\"></div>
    <div class=\"pe-shadow__top pe-shadow--z-1\"></div>
  </div>
  <div class=\"pe-ripple pe-ripple--constrained\">
    <div class=\"pe-ripple__mask\">
      <div class=\"pe-ripple__waves\"></div>
    </div>
  </div>
  <div class=\"pe-button__wash\"></div>
  <div class=\"pe-button__focus\"></div>
  <div class=\"pe-button__label\">
    Themed button
  </div>
</div></a>
"
`;

exports[`test Themed button (with option disabled) 1`] = `
"<a tabindex=\"-1\" disabled class=
\"pe-button pe-text-button pe-raised-button pe-button--disabled tests-raised-button-themed-button\">
<div class=\"pe-button__content\">
  <div class=\"pe-button__label\">
    Disabled themed button
  </div>
</div></a>
"
`;
